src/pilot.c

```diff
@@ -2058,6 +2060,41 @@ void pilot_update( Pilot* pilot, double dt )
          o->rtimer = MIN( o->rtimer, reload_time );
       }
 
+      /* Get the asteroid target. */
+      if (pilot->nav_asteroid != -1) {
+         field = &cur_system->asteroids[pilot->nav_anchor];
+         ast = &field->asteroids[pilot->nav_asteroid];
+      }
+
+      /* Special case of beam weapons with weapon set only_inrange: turn off if needed. */
+      if (outfit_isBeam(o->outfit)) {
```

Comment:
Wouldn't this part make more sense in `think_beam` in `src/weapon.c`? You can do the same thing as when energy is out by setting `w->timer = -1;` to stop the beam.

Reply:
Yes, it works better, and without the bug that suppresses all other weapons.
I'll re-write history (because it's really messy), test, and push in a separate branch for now.